### ISSUE TITLE
🐛 Bug Report: When signin resolver fails using enableExperimentalRedirectFlow, user sees blank page.

### ISSUE THREAD

**Issue by JWWilks** [backstage/backstage#24856]
### 📜 Description

Errors from the sign-in process aren't shown to the user when using `enableExperimentalRedirectFlow`, and instead the /handler/frame returns before redirecting the user, resulting in a seemingly blank page.

### 👍 Expected behavior

When signing in with `experimentalRedirectFlow` enabled, if there is an error in the sign in process (eg. "Auth response is missing cookie nonce", "No redirectUrl provided in request query parameters", or a different error thrown intentionally by the SignIn resolver), the /handler/frame page should still send the user back to the sign-in screen and the error should be shown to the user.

This should a similar experience to the popup redirect flow - where the popup window is closed by the sign-in page and the user is shown the error on the SignIn page - except with a redirect.

### 👎 Actual Behavior with Screenshots

The user is left on a blank /handler/frame page and not redirected to the redirect url. The frame is rendered with the code intended sending the error response to the popup. No errors are shown to the user.

![image](https://github.com/backstage/backstage/assets/75726489/ca2559e2-2d9b-45b5-9996-ce2b99303339)

### 👟 Reproduction steps

1. Enable enableExperimentalRedirectFlow in config.
2. In your custom SignIn resolver (if you have one), intentionally throw an error immediately. Eg "User not found". Alternatively, sign in with an account on the provider which does not map via any of your resolvers to a user in your backstage catalog.
3. Clear session and login to backstage using the SignIn provider.
4. After successfully authenticating with the provider but not the resolver, you will be sent to a seemingly blank /handler/frame page with the underlying source code for the popup auth, and not redirected to the redirectUrl.

#### The Problem
I believe the reason this happens is because the error is simply not handled when using a redirect flow.  The only condition under which the redirect flow can successfully redirect is if no errors are thrown:

https://github.com/backstage/backstage/blob/2c3f493ee32d67d3a300d13e73e352f91145a3ee/plugins/auth-node/src/oauth/createOAuthRouteHandlers.ts#L229-L239

The catch block around the route handler contains code to use `postMessage` to send the error to the window's parent, but I believe the JS in `sendWebMessageResponse` will have no effect on the window unless it's a popup window.

https://github.com/backstage/backstage/blob/2c3f493ee32d67d3a300d13e73e352f91145a3ee/plugins/auth-node/src/oauth/createOAuthRouteHandlers.ts#L246-L256

#### Possible Solutions
I believe an ideal solution would either:
- append `?error=errormessage` to the redirect url and still redirect the user, then handle this on the frontend. (security implications, messy)
- add the error message to a cookie or local storage, still redirect the user, then handle this on the frontend. (security implications, still messy)
- render a friendly error message page on the backend's /handler/frame endpoint before sending them back to the frontend with a delayed redirect. (easier to implement, but wont necessarily respect frontend styling and won't be a good UX if the user doesn't notice the redirect)

### 🖥️ Your Environment

- **Browser Machine(s)**: Chrome & Edge on two windows machines
- **Environment(s)**: Locally on windows & in deployed k8s pod
- **Auth Provider**: Azure Entra
- **Auth resolver**: Custom resolver mapping entra names to catalog entity names

<details>
  <summary>yarn backstage-cli info</summary>
  
```
OS:   Windows_NT 10.0.19045 - win32/x64
node: v20.11.1
yarn: 1.22.19
cli:  0.26.5 (installed)
backstage:  1.27.0

Dependencies:
  @backstage/app-defaults                                          1.5.5
  @backstage/backend-app-api                                       0.5.14, 0.7.2, 0.7.3
  @backstage/backend-common                                        0.20.2, 0.21.7, 0.22.0
  @backstage/backend-defaults                                      0.2.18
  @backstage/backend-dev-utils                                     0.1.4
  @backstage/backend-openapi-utils                                 0.1.11
  @backstage/backend-plugin-api                                    0.6.17, 0.6.18
  @backstage/backend-tasks                                         0.5.22, 0.5.23
  @backstage/catalog-client                                        1.6.4, 1.6.5
  @backstage/catalog-model                                         1.4.5, 1.5.0
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.5
  @backstage/cli                                                   0.26.5
  @backstage/config-loader                                         1.8.0
  @backstage/config                                                1.2.0
  @backstage/core-app-api                                          1.12.5
  @backstage/core-compat-api                                       0.2.5
  @backstage/core-components                                       0.13.10, 0.14.7
  @backstage/core-plugin-api                                       1.9.2
  @backstage/dev-utils                                             1.0.32
  @backstage/e2e-test-utils                                        0.1.1
  @backstage/errors                                                1.2.4
  @backstage/eslint-plugin                                         0.1.8
  @backstage/frontend-plugin-api                                   0.6.5
  @backstage/integration-aws-node                                  0.1.12
  @backstage/integration-react                                     1.1.27
  @backstage/integration                                           1.10.0, 1.11.0
  @backstage/plugin-api-docs                                       0.11.5
  @backstage/plugin-app-backend                                    0.3.66
  @backstage/plugin-app-node                                       0.1.18
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.10
  @backstage/plugin-auth-backend-module-aws-alb-provider           0.1.10
  @backstage/plugin-auth-backend-module-azure-easyauth-provider    0.1.1
  @backstage/plugin-auth-backend-module-bitbucket-provider         0.1.1
  @backstage/plugin-auth-backend-module-cloudflare-access-provider 0.1.1
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.13
  @backstage/plugin-auth-backend-module-github-provider            0.1.15
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.15
  @backstage/plugin-auth-backend-module-google-provider            0.1.15
  @backstage/plugin-auth-backend-module-guest-provider             0.1.4
  @backstage/plugin-auth-backend-module-microsoft-provider         0.1.13
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.15
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.11
  @backstage/plugin-auth-backend-module-oidc-provider              0.1.9
  @backstage/plugin-auth-backend-module-okta-provider              0.0.11
  @backstage/plugin-auth-backend                                   0.22.5
  @backstage/plugin-auth-node                                      0.4.13
  @backstage/plugin-auth-react                                     0.1.2
  @backstage/plugin-catalog-backend-module-gitlab-org              0.0.1
  @backstage/plugin-catalog-backend-module-gitlab                  0.3.15
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.16
  @backstage/plugin-catalog-backend                                1.22.0
  @backstage/plugin-catalog-common                                 1.0.23
  @backstage/plugin-catalog-graph                                  0.4.5
  @backstage/plugin-catalog-import                                 0.11.0
  @backstage/plugin-catalog-node                                   1.11.1, 1.12.0
  @backstage/plugin-catalog-react                                  1.11.3, 1.12.0
  @backstage/plugin-catalog                                        1.20.0
  @backstage/plugin-events-node                                    0.3.4
  @backstage/plugin-home-react                                     0.1.12, 0.1.13
  @backstage/plugin-home                                           0.7.4
  @backstage/plugin-org                                            0.6.25
  @backstage/plugin-permission-backend-module-allow-all-policy     0.1.15
  @backstage/plugin-permission-backend                             0.5.42
  @backstage/plugin-permission-common                              0.7.13
  @backstage/plugin-permission-node                                0.7.28, 0.7.29
  @backstage/plugin-permission-react                               0.4.22
  @backstage/plugin-proxy-backend                                  0.4.16
  @backstage/plugin-scaffolder-backend-module-azure                0.1.10
  @backstage/plugin-scaffolder-backend-module-bitbucket-cloud      0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.1.8
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.2.8
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.10
  @backstage/plugin-scaffolder-backend-module-gitea                0.1.8
  @backstage/plugin-scaffolder-backend-module-github               0.2.8
  @backstage/plugin-scaffolder-backend-module-gitlab               0.4.0
  @backstage/plugin-scaffolder-backend                             1.22.6
  @backstage/plugin-scaffolder-common                              1.5.2
  @backstage/plugin-scaffolder-node                                0.4.4
  @backstage/plugin-scaffolder-react                               1.8.5
  @backstage/plugin-scaffolder                                     1.20.0
  @backstage/plugin-search-backend-module-catalog                  0.1.24
  @backstage/plugin-search-backend-module-pg                       0.5.27
  @backstage/plugin-search-backend-module-stack-overflow-collator  0.1.11
  @backstage/plugin-search-backend-module-techdocs                 0.1.23
  @backstage/plugin-search-backend-node                            1.2.22
  @backstage/plugin-search-backend                                 1.5.8
  @backstage/plugin-search-common                                  1.2.11
  @backstage/plugin-search-react                                   1.7.11
  @backstage/plugin-search                                         1.4.11
  @backstage/plugin-techdocs-backend                               1.10.5
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.10
  @backstage/plugin-techdocs-node                                  1.12.4
  @backstage/plugin-techdocs-react                                 1.2.4
  @backstage/plugin-techdocs                                       1.10.5
  @backstage/plugin-user-settings                                  0.8.6
  @backstage/release-manifests                                     0.0.11
  @backstage/repo-tools                                            0.9.0
  @backstage/test-utils                                            1.5.5
  @backstage/theme                                                 0.4.4, 0.5.4
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.8
``` 
</details> 

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by stephenglass** [comment]
Created PR #25823 with a fix! 😀

**Comment by makii42** [comment]
Sweet, we ran into this as well, and were planning to contribute the third option back that we implemented in a dirty way:

> render a friendly error message page on the backend's /handler/frame endpoint before sending them back to the frontend with a delayed redirect. (easier to implement, but wont necessarily respect frontend styling and won't be a good UX if the user doesn't notice the redirect)

Seems we can ditch that after our next version bump, Thanks <3

**Comment by vinzscam** [comment]
Closed by #25823
